File: dailyQuestion/2020/2020-06/06-30/golang/solution.go
package main

import "fmt"

type CQueue struct {
	stack_in  []int
	stack_out []int
}

func Constructor() CQueue {
	return CQueue{stack_in: []int{}, stack_out: []int{}}
}

func (this *CQueue) AppendTail(value int) {
	this.stack_in = append(this.stack_in, value)
}

func (this *CQueue) DeleteHead() int {
	if len(this.stack_out) == 0 {
		for len(this.stack_in) != 0 {
			item := this.stack_in[len(this.stack_in)-1]
			this.stack_out = append(this.stack_out, item)
			this.stack_in = this.stack_in[0 : len(this.stack_in)-1]
		}
	}
	if len(this.stack_out) != 0 {
		item := this.stack_out[len(this.stack_out)-1]
		this.stack_out = this.stack_out[0 : len(this.stack_out)-1]
		return item
	}
	return -1
}

func main() {
	obj := Constructor()
	obj.AppendTail(1)
	obj.AppendTail(2)
	obj.AppendTail(3)
	fmt.Println(obj.DeleteHead())
	fmt.Println(obj.DeleteHead())
	obj.AppendTail(4)
	fmt.Println(obj.DeleteHead())
	fmt.Println(obj.DeleteHead())
	fmt.Println(obj.DeleteHead())

}

File: dailyQuestion/2020/2020-06/06-30/python/solution.py
class CQueue:

    def __init__(self):
        self.stack_in = []
        self.stack_out = []

    def appendTail(self, value: int) -> None:
        self.stack_in.append(value)

    def deleteHead(self) -> int:
        if len(self.stack_out) == 0:
            while self.stack_in:
                self.stack_out.append(self.stack_in.pop())
        if len(self.stack_out) != 0:
            return self.stack_out.pop()
        return -1

if __name__ == "__main__" :
    obj = CQueue()
    obj.appendTail(1)
    obj.appendTail(2)
    obj.appendTail(3)
    assert obj.deleteHead() == 1
    assert obj.deleteHead() == 2
    obj.appendTail(4)
    assert obj.deleteHead() == 3
    assert obj.deleteHead() == 4
    assert obj.deleteHead() == -1